crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -395,11 +395,11 @@ impl CheckpointContents {
 // CheckpointBuilder::split_checkpoint_chunks should also be updated accordingly.
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
 pub struct FullCheckpointContents {
-    transactions: Vec<ExecutionData>,
+    pub transactions: Vec<ExecutionData>,
```

Comment:
is it necessary to make these `pub`/why?

for transactions, can you use the `iter` method or make an `into_iter` method if you want one?

for `user_signautres`, I don't see any use outside this file?